File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/preset/InventoryPresetItemUpdateRequestMessage.scala





















// Generated on 05/03/2013 10:34:58
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InventoryPresetItemUpdateRequestMessage extends MessageDeserializer {
  val messageId = 6210

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InventoryPresetItemUpdateRequestMessage.messageId

    val presetId: Byte
    val position: UByte
    val objUid: Int

    def serialize(buf: Buffer) {
      buf.writeByte(presetId)
      buf.writeUByte(position)
      buf.writeInt(objUid)
    }

  }

  protected def create(presetId: Byte, position: UByte, objUid: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val presetId = buf.readByte()
    val position = buf.readUByte()
    val objUid = buf.readInt()

    create(presetId, position, objUid)
  }

}

object InventoryPresetItemUpdateRequestMessage extends InventoryPresetItemUpdateRequestMessage {
  case class MessageImpl(presetId: Byte, position: UByte, objUid: Int) extends Message
  type Target = MessageImpl

  override def create(presetId: Byte, position: UByte, objUid: Int) = MessageImpl(presetId, position, objUid)

  def apply(presetId: Byte, position: UByte, objUid: Int) = create(presetId, position, objUid)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InventoryPresetItemUpdateRequestMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InventoryPresetItemUpdateRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6210;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte presetId;
    public short position;
    public int objUid;
    
    public InventoryPresetItemUpdateRequestMessage() { }
    
    public InventoryPresetItemUpdateRequestMessage(byte presetId, short position, int objUid) {
        this.presetId = presetId;
        this.position = position;
        this.objUid = objUid;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(presetId);
        buf.writeUByte(position);
        buf.writeInt(objUid);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        presetId = buf.readByte();
        if (presetId < 0)
            throw new RuntimeException("Forbidden value on presetId = " + presetId + ", it doesn't respect the following condition : presetId < 0");
        position = buf.readUByte();
        if (position < 0 || position > 255)
            throw new RuntimeException("Forbidden value on position = " + position + ", it doesn't respect the following condition : position < 0 || position > 255");
        objUid = buf.readInt();
        if (objUid < 0)
            throw new RuntimeException("Forbidden value on objUid = " + objUid + ", it doesn't respect the following condition : objUid < 0");
    }
    
}
